binary_classification/src/lib.rs

```diff
@@ -1,90 +1,156 @@
+        .parse::<f32>()
+        .map_err(|e| InvalidArgument::invalid_value("threshold", e))
+}
 
-    fn transform(&mut self, input: Tensor<f32>) -> Tensor<u32> {
-        // let value = input.into();
-        let mut label: u32 = 0;
-        if input.elements() > &[self.threshold] {
-            label = 1
+impl InvalidArgument {
+    fn not_found(name: impl Into<String>) -> Self {
+        InvalidArgument {
+            name: name.into(),
+            reason: BadArgumentReason::NotFound,
         }
-        let v: Vec<u32> = vec![label];
+    }
 
-        Tensor::new_vector(v)
+    fn invalid_value(name: impl Into<String>, reason: impl Display) -> Self {
+        InvalidArgument {
+            name: name.into(),
+            reason: BadArgumentReason::InvalidValue(reason.to_string()),
+        }
     }
 }
 
-#[cfg(feature = "metadata")]
-pub mod metadata {
-    wit_bindgen_rust::import!("../wit-files/rune/runtime-v1.wit");
-    wit_bindgen_rust::export!("../wit-files/rune/rune-v1.wit");
-
-    struct RuneV1;
-
-    impl rune_v1::RuneV1 for RuneV1 {
-        fn start() {
-            use runtime_v1::*;
-
-            let metadata = Metadata::new(
-                "Binary Classification",
-                env!("CARGO_PKG_VERSION"),
-            );
-            metadata.set_description(
-                "Classify each element in a tensor depending on whether they are above or below a certain threshold.",
-            );
-            metadata.set_repository(env!("CARGO_PKG_REPOSITORY"));
-            metadata.set_homepage(env!("CARGO_PKG_HOMEPAGE"));
-            metadata.add_tag("classify");
-
-            let threshold = ArgumentMetadata::new("threshold");
-            threshold.set_default_value("0.5");
-            threshold.set_description("The classification threshold.");
-            threshold.set_type_hint(TypeHint::Float);
-            metadata.add_argument(&threshold);
-
-            let input = TensorMetadata::new("input");
-            input.set_description("The numbers to classify");
-            let hint = supported_shapes(
-                &[ElementType::Float32],
-                Dimensions::Fixed(&[0]),
-            );
-            input.add_hint(&hint);
-            metadata.add_input(&input);
-
-            let output = TensorMetadata::new("classified");
-            output.set_description("A tensor of `1`'s and `0`'s, where `1` indicates an element was above the `threshold` and `0` means it was below.");
-            let hint = supported_shapes(
-                &[ElementType::Uint32],
-                Dimensions::Fixed(&[0]),
-            );
-            output.add_hint(&hint);
-            metadata.add_output(&output);
-
-            register_node(&metadata);
-        }
+fn transform(input: &[f32], threshold: f32) -> Vec<u32> {
+    // let value = input.into();
+    let mut label: u32 = 0;
+    if input > &[threshold] {
+        label = 1
     }
+    let v: Vec<u32> = vec![label];
```

Comment:
This will give wrong results if you pass in an input that contains more than one element. What about accepting an `ArrayViewD<f32>` and doing something like `input.mapv(|value| if value > threshold { 1 } else { 0 })` to apply the threshold to each element in your N-dimensional input?

This also lets you accept a dynamic number of dimensions instead of `DimensionsParam::Fixed(&[0])`.